File: Java/lista_1/src/WierszTrojkataPascala.java
public class WierszTrojkataPascala {
    private final int[] tablica;

    public WierszTrojkataPascala(int n) {
        if (n < 0) {
            throw new IllegalArgumentException(n + " - Nieprawidłowy numer wiersza");
        }
        this.tablica = obliczenieNtegoWiersza(n);
    }

    public int MtyElementWiersza(int m) {
        if (m < 0 || m >= tablica.length) {
            throw new IndexOutOfBoundsException(m + " - liczba spoza zakresu");
        }
        return tablica[m];
    }

    private int[] obliczenieNtegoWiersza(int n) {
        int[] wiersz = new int[n + 1];

        if (n == 0) {
            wiersz[0] = 1;
            return wiersz;
        }

        if (n == 1) {
            wiersz[0] = 1;
            wiersz[1] = 1;
            return wiersz;
        }

        int[] poprzedni_wiersz = new WierszTrojkataPascala(n - 1).getTablica();

        for (int i = 0; i <= n; i++) {
            if (i == 0 || i == n) {
                wiersz[i] = 1;
            } else {
                wiersz[i] = poprzedni_wiersz[i - 1] + poprzedni_wiersz[i];
            }
        }
        return wiersz;
    }

    public int[] getTablica() {
        return tablica;
    }
}

File: Java/lista_4/src/lista_1/WierszTrojkataPascala.cpp
#include "WierszTrojkataPascala.h"

using namespace std;

WierszTrojkataPascala::WierszTrojkataPascala(int n) {
    if (n < 0) {
        throw invalid_argument(to_string(n) + " - nieprawidłowy numer wiersza");
    }
    size = n + 1;
    tablica = new int[size];
    obliczenieNtegoWiersza(n);
}

WierszTrojkataPascala::~WierszTrojkataPascala() {
    delete[] tablica;
}

int WierszTrojkataPascala::MtyElementWiersza(int m) {
    if (m < 0 || m >= size) {
        throw out_of_range(to_string(m) + " - liczba spoza zakresu");
    }
    return tablica[m];
}

void WierszTrojkataPascala::obliczenieNtegoWiersza(int n) {
    if (n == 0) {
        tablica[0] = 1;
        return;
    }

    if (n == 1) {
        tablica[0] = 1;
        tablica[1] = 1;
        return;
    }

    WierszTrojkataPascala poprzedni_wiersz(n - 1);

    for (int i = 0; i <= n; ++i) {
        if (i == 0 || i == n) {
            tablica[i] = 1;
        } else {
            tablica[i] = poprzedni_wiersz.tablica[i - 1] + poprzedni_wiersz.tablica[i];
        }
    }
}